packages/hebao_v1/contracts/modules/dapps/base/GenericDAppModule.sol

```diff
@@ -0,0 +1,124 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.13;
+pragma experimental ABIEncoderV2;
+
+import "../../../thirdparty/BytesUtil.sol";
+
+import "../../security/SecurityModule.sol";
+
+
+/// @title GenericDAppModule
+/// @dev GenericDAppModule allows wallet owners to transact directly or through meta
+///      transactions on the specified dApp. The transaction data must be appended
+///      with the wallet address.
+contract GenericDAppModule is SecurityModule
+{
+    using BytesUtil for bytes;
+
+    address public dapp;
+    string  public name;
+
+    constructor(
+        Controller    _controller,
+        address       _dapp,
+        string memory _name
+        )
+        public
+        SecurityModule(_controller)
+    {
+        require(
+            _dapp != address(0) &&
+            !controller.moduleRegistry().isModuleRegistered(_dapp) &&
+            !controller.walletRegistry().isWalletRegistered(_dapp),
```

Comment:
Any particular reason to check for these? It doesn't make sense to use those addresses for sure, but these addresses should not be more dangerous than other addresses.

Reply:
I think we can remove these checks :)